packer/builder/azure/arm/deployment_poller.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"time"
+)
+
+const (
+	DeployCanceled  = "Canceled"
+	DeployFailed    = "Failed"
+	DeployDeleted   = "Deleted"
+	DeploySucceeded = "Succeeded"
+)
+
+type DeploymentPoller struct {
+	getProvisioningState func() (string, error)
+	pause                func()
+}
+
+func NewDeploymentPoller(getProvisioningState func() (string, error)) *DeploymentPoller {
+	pollDuration := time.Second * 15
+
+	return &DeploymentPoller{
+		getProvisioningState: getProvisioningState,
+		pause:                func() { time.Sleep(pollDuration) },
```

Comment:
you probably need to change this to `select` with `time.After` to support cancellation at some point in time... maybe make it cancellable now? (Yes, I know that's broken in the smapi version... :-( )